labgrid/driver/shelldriver.py

```diff
@@ -565,3 +567,26 @@ def get_default_interface_device_name(self, version=4):
             raise ExecutionError(f"Multiple IPv{version} default routes found")
 
         return matches[0]
+
+    @Driver.check_active
+    def get_ip_interfaces(self, device=None):
+        """ Retrieves IP interfaces for given interface name.
+
+        Args:
+            device (str): Name of the interface to query, defaults to default route's device name.
+
+        Returns:
+            List of IPv4Interface or IPv6Interface objects
+        """
+        if device is None:
+            device = self.get_default_interface_device_name()
+
+        regex = r"""\d+: # leading number
+                \s+\w+ # interface name
+                \s+inet6?\s+(\S+) # IP address, prefix
+                .*global # global scope, not host scope"""
+
+        ip_show = self._run_check(f"ip -o addr show dev {device}")
```

Comment:
You can output directly parsable JSON with `ip -j addr`. I've done something similar to this before with https://github.com/JPEWdev/yocto-doom-demo/blob/main/ci/conftest.py#L29, but in that case I was assigning the address specified in the `NetworkResource` if it wasn't already assigned.

Reply:
Unfortunately busybox' `ip` does not support `-j`, so this is not an option.